engine/src/main/scala/cromwell/engine/workflow/WorkflowManagerActor.scala

```diff
@@ -342,11 +342,16 @@ class WorkflowManagerActor(config: Config)
 
     import PromiseActor.EnhancedActorRef
 
+    val shadowExecutionEnabled = config.getBoolean("system.shadowExecutionEnabled")
+
     val message  = MaterializeWorkflowDescriptorActor.MaterializeWorkflow(workflowId, source, config)
     val materializeWorkflowDescriptorActor = context.actorOf(MaterializeWorkflowDescriptorActor.props())
     materializeWorkflowDescriptorActor.askNoTimeout(message) map {
       case MaterializationSuccess(descriptor) =>
-        val wfActor = context.actorOf(WorkflowActor.props(descriptor), s"WorkflowActor-$workflowId")
+        val wfActor = shadowExecutionEnabled match {
+          case true => context.actorOf(ShadowWorkflowActor.props(), s"WorkflowActor-$workflowId")
```

Comment:
I like the idea of telling a workflow what it's ID will be. Sources will certainly be passed in here, maybe part of the set-up FSM or Validation ticket?